app/javascript/controllers/select_with_auto_complete_controller.js

```diff
@@ -0,0 +1,484 @@
+      case "ArrowDown":
+        if (this.#filteredOptions.length > 0) {
+          if (altKey && this.#isClosed()) {
+            this.#open();
+          } else {
+            if (this.#isClosed()) {
+              this.#open();
+            }
+            if (this.#filteredOptions.length > 1) {
+              this.#setOption(this.#getNextOption(this.#option));
+            } else {
+              this.#setOption(this.#firstOption);
+            }
+          }
+        }
+        flag = true;
+        break;
+
+      case "Up":
+      case "ArrowUp":
+        if (this.#hasOptions()) {
+          if (this.#isOpen()) {
+            this.#setOption(this.#getPreviousOption(this.#option));
+          } else {
+            this.#open();
+            if (!altKey) {
+              this.#setOption(this.#lastOption);
+            }
+          }
+        }
+        flag = true;
+        break;
+
+      case "Esc":
+      case "Escape":
+        if (this.#isOpen()) {
+          this.#close();
+        } else {
+          this.#setValue();
+          this.#setOption(null);
+        }
+        flag = true;
+        break;
+
+      case "Tab":
+        if (this.comboboxTarget.value) {
+          this.#filter = this.comboboxTarget.value;
+          const option = this.#filterOptions();
+          this.#setValue(option);
+          this.#setOption(option);
+        }
+        this.#close();
+        break;
+
+      case "Home":
+        this.comboboxTarget.setSelectionRange(0, 0);
+        flag = true;
+        break;
+
+      case "End":
+        const length = this.comboboxTarget.value.length;
+        this.comboboxTarget.setSelectionRange(length, length);
+        flag = true;
+        break;
+
+      default:
+        break;
+    }
+
+    if (flag) {
+      event.stopPropagation();
+      event.preventDefault();
+    }
+  }
+
+  #onComboboxKeyUp(event) {
+    let flag = false;
+    const char = event.key;
+
+    if (isPrintableCharacter(char)) {
+      this.#filter += char;
+    }
+
+    if (event.key === "Escape" || event.key === "Esc") {
+      return;
+    }
+
+    switch (event.key) {
+      case "Backspace":
+        this.#filter = this.comboboxTarget.value;
+        this.debouncedFilterAndUpdate();
+        flag = true;
+        break;
+
+      case "Left":
+      case "ArrowLeft":
+      case "Right":
+      case "ArrowRight":
+      case "Home":
+      case "End":
+        this.#setOption(null);
+        flag = true;
+        break;
+
+      default:
+        if (isPrintableCharacter(char)) {
+          this.debouncedFilterAndUpdate();
+          flag = true;
+        }
+
+        break;
+    }
+
+    if (flag) {
+      event.stopPropagation();
+      event.preventDefault();
+    }
+  }
+
+  #onComboboxClick() {
+    if (this.#isOpen()) {
+      this.#close();
+    } else {
+      this.#open();
+    }
+  }
+
+  #onComboboxFocus() {
+    this.#filter = this.comboboxTarget.value;
+    this.#filterOptions();
+  }
+
+  #onComboboxBlur() {
```

Comment:
Make sure your invalid types are cleared on blur.
To do this, update this method to: 

```javascript
#onComboboxBlur() {
  this.#filter = this.comboboxTarget.value;
  const option = this.#filterOptions();
  this.#setValue(option);
  this.#setOption(option);
  this.#close();
}
```